source/coll_db.f90

```diff
@@ -586,6 +586,16 @@ subroutine cdb_writeDB_newFromOld
     return
   end if
 
+  do j=1,cdb_nColl
+    ! Fixing family setting to that of the first collimator in the family when writing old CollDB in new format
+    ! Allows to take into account if settings were defined in fort.3 or old CollDB
+    if(j > 1 .and. cdb_cFamily(j) == cdb_cFamily(j-1)) then
```

Comment:
Here you assume that collimators in the same family are all contiguous in memory. I guess that this is not the case e.g. for IR TCTs... Then you would store in memory the settings of the first collimator in the last batch of consecutive collimators in the same family

Reply:
The logic of the family settings reconstruction has been reworked in a separate subroutine. As you suggested in your comments, the reconstruction is now based on the most frequent setting for each family.